GeneralsMD/Code/GameEngine/Source/GameClient/GUI/GUICallbacks/Menus/ReplayMenu.cpp

```diff
@@ -102,6 +102,75 @@ UnicodeString GetReplayFilenameFromListbox(GameWindow *listbox, Int index)
 	return fname;
 }
 
+//-------------------------------------------------------------------------------------------------
+
+static Bool readReplayMapInfo(const AsciiString& filename, RecorderClass::ReplayHeader &header, ReplayGameInfo &info, const MapMetaData *&mapData)
+{
+	header.forPlayback = FALSE;
+	header.filename = filename;
+
+	if (TheRecorder != NULL && TheRecorder->readReplayHeader(header))
+	{
+		if (ParseAsciiStringToGameInfo(&info, header.gameOptions))
+		{
+			if (TheMapCache != NULL)
+				mapData = TheMapCache->findMap(info.getMap());
+			else
+				mapData = NULL;
+
+			return true;
+		}
+	}
+	return false;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+static void removeReplayExtension(UnicodeString& replayName)
+{
+	const Int extensionLength = TheRecorder->getReplayExtention().getLength();
+		for (Int k=0; k < extensionLength; ++k)
+			replayName.removeLastChar();
+}
+
+//-------------------------------------------------------------------------------------------------
+
+static UnicodeString createReplayName(const AsciiString& filename)
+{
+	AsciiString lastReplayFName = TheRecorder->getLastReplayFileName();
+	lastReplayFName.concat(TheRecorder->getReplayExtention());
+	UnicodeString replayName;
+
+	if (lastReplayFName.compareNoCase(filename) == 0)
+	{
+		replayName = TheGameText->fetch("GUI:LastReplay");
+	}
+	else
+	{
+		replayName.translate(filename);
+		removeReplayExtension(replayName);
+	}
+	return replayName;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+static UnicodeString createMapName(const AsciiString& filename, const ReplayGameInfo& info, const MapMetaData *mapData)
+{
+	UnicodeString mapName;
+	if (!mapData)
+	{
+		// TheSuperHackers @bugfix helmutbuhler 08/03/2025 Just use the filename.
+		// Displaying a long map path string would break the map list gui.
+		const char* filename = info.getMap().reverseFind('\\');
```

Comment:
Note to myself: I suspect this will still break the Replay list if the file name is very long then. So we perhaps need another change that caps the length of the Replay name here. Unrelated to this Pull Request.